src/edu/uw/easysrl/syntax/parser/ParserBuilder.java

```diff
@@ -145,9 +141,11 @@ public double getNBestBeam() {
 	private CutoffsDictionaryInterface cutoffs;
 	private boolean useSupertaggedInput = false;
 	private final List<Combinator> combinators = new ArrayList<>(Combinator.STANDARD_COMBINATORS);
-	private int maxChartSize;
+	private int maxChartSize = 300000;
+	private int maxAgendaSize = 99999999;
 	private NormalForm normalForm = new NormalForm();
-	private double nBestBeam = 0.001;
+	private double nbestBeam = 0.001;
+	private List<ParserListener> listeners = new ArrayList<>();
```

Comment:
Could you swap this to Collections.emptyList()? It should perform a lot better, because when you do a for loop over it, ArrayList has to allocate a new iterator.